sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -170,9 +170,8 @@ trait ProgressReporter extends Logging {
       )
     }
 
-    val sinkProgress = SinkProgress(
-      sink.toString,
-      sinkCommitProgress.map(_.numOutputRows))
+    val sinkOutput = if (hasExecuted) sinkCommitProgress.map(_.numOutputRows) else Some(0L)
```

Comment:
its not clear to me why this needs to be done. MicroBatchExecution.runBatch sets this `sinkCommitProgress` variable after every batch, even if the batch is empty. So I would expect this progress to have update metrics, which would naturally be 0 for an empty batch that did not produce anything.

Reply:
but in this case, we're not calling runBatch, because there was no new data and no zero-data microbatch